### DISCUSSION TITLE
add_subdirectory given source "nlohmann_json" which is not an existing directory.

### DISCUSSION THREAD

**Discussion by hygxy** [nlohmann/json#2921] [Q&A] [answered]
I want to use this repository in "Embedded" mode using CMake.  I did exactly what is described in README(place the entire source tree in a subdirectory and call add_subdirectory() in your CMakeLists.txt file), but get the following error:
```
"add_subdirectory given source "nlohmann_json" which is not an existing directory.“ 
```

Besides, I would like to know what does "..." means in line ```add_library(foo ...)```

**Comment by nlohmann** [comment, marked as answer]
See https://github.com/nlohmann/json/tree/develop/test/cmake_add_subdirectory/project for an example.
